src/subtitles-octopus.js

```diff
@@ -414,28 +427,47 @@ var SubtitlesOctopus = function (options) {
         self.rafId = 0;
     }
 
+    /**
+     * Compare rects with a given relative (to 2nd rect) threshold.
+     * @param {Object} rect1 - Rect 1.
+     * @param {number} rect1.width - Rect 1 width.
+     * @param {number} rect1.height - Rect 2 height.
+     * @param {Object} rect2 - Rect 2.
+     * @param {number} rect2.width - Rect 2 width.
+     * @param {number} rect2.height - Rect 2 height.
+     * @param {number} threshold - Relative threshold.
+     * @returns {boolean} Rects are equal (or not) within a given relative (to 2nd rect) threshold.
+     */
+    function fuzzyEqualRect(rect1, rect2, threshold) {
+        return (Math.abs(rect1.width - rect2.width) < threshold * rect2.width &&
+            Math.abs(rect1.height - rect2.height) < threshold * rect2.height);
+    }
+
     self.resetRenderAheadCache = function (isResizing) {
         if (self.renderAhead > 0) {
             var newCache = [];
             if (isResizing && self.oneshotState.prevHeight && self.oneshotState.prevWidth) {
                 if (self.oneshotState.prevHeight == self.canvas.height &&
                     self.oneshotState.prevWidth == self.canvas.width) return;
-                var timeLimit = 10, sizeLimit = self.renderAhead * 0.3;
-                if (self.canvas.height >= self.oneshotState.prevHeight * (1.0 - self.resizeVariation) &&
-                    self.canvas.height <= self.oneshotState.prevHeight * (1.0 + self.resizeVariation) &&
-                    self.canvas.width >= self.oneshotState.prevWidth * (1.0 - self.resizeVariation) &&
-                    self.canvas.width <= self.oneshotState.prevWidth * (1.0 + self.resizeVariation)) {
-                    console.debug('viewport changes are small, leaving more of prerendered buffer');
-                    timeLimit = 30;
-                    sizeLimit = self.renderAhead * 0.5;
-                }
-                var stopTime = self.video.currentTime + self.timeOffset + timeLimit;
+
+                var tm = self.video.currentTime + self.timeOffset;
+                var minTime = tm + self.renderAheadResetMinTime;
+                var maxTime = self.renderAheadResetMaxTime > 0 ? tm + self.renderAheadResetMaxTime : Infinity;
+                var sizeLimit = self.renderAhead * self.renderAheadResetMaxFill;
                 var size = 0;
+
                 for (var i = 0; i < self.renderedItems.length; i++) {
                     var item = self.renderedItems[i];
-                    if (item.emptyFinish < 0 || item.emptyFinish >= stopTime) break;
+
+                    if (item.emptyFinish < 0 || item.eventFinish >= maxTime) break;
+
+                    if (item.eventFinish >= minTime &&
+                        !fuzzyEqualRect(item.viewport, self.canvas, self.resizeVariation)) continue;
```

Comment:
With `continue` the far good frames take up some size, leaving less room for the near frames and making a gap.
This can happen with smooth resizing, when the far frames are more recent and correspond to the current canvas size.
Using `break` forces to rerender the good frames.